app/components/data/details/data-container-files.component.ts

```diff
@@ -64,4 +63,47 @@ export class DataContainerFilesComponent implements OnDestroy {
             return observable;
         });
     }
+
+    @autobind()
+    public handleFileDelete(files: File[]) {
+        const fileCount = files.length;
+        const taskTitle = `Delete ${fileCount} files from ${this.container.name}`;
+
+        return this.backgroundTaskService.startTask(taskTitle, (task) => {
+            let deleted = 0;
+            const blobCache = this.storageService.getBlobFileCache({
+                container: this.container.id,
+            });
+
+            // NOTE: slight pause in-between deletes to ease load on storage account
+            // may or may not be a great idea.
+            const observable = Observable.interval(100).take(fileCount);
+            observable.subscribe({
+                next: (i) => {
+                    deleted++;
+                    return this.storageService.deleteBlobIfExists(this.container.id, files[i].name).subscribe({
+                        next: (response) => {
+                            task.name.next(`${taskTitle} (${deleted}/${fileCount})`);
+                            task.progress.next(deleted / fileCount * 100);
+
+                            if (response && blobCache) {
+                                blobCache.deleteItem(files[i]);
```

Comment:
shouldn't it be the deleteBlobIfexists that remove from the cache. I feel this would be better. THat way you don't have to do that everytime

Reply:
on a .subscribe(() => { /* remove from cache */ })?

probably a very good idea and one that i am not sure why i didn't think of :)